src/videostreamidset/DefaultVideoStreamIdSet.ts

```diff
@@ -66,4 +66,9 @@ export default class DefaultVideoStreamIdSet implements VideoStreamIdSet {
   toJSON(): number[] {
     return this.array();
   }
+
+  truncate(length: number): DefaultVideoStreamIdSet {
+    const x: number[] = this.array();
+    return new DefaultVideoStreamIdSet(x.splice(0, length));
```

Comment:
Should we log a warning that we will truncate since there are more video to subscribe than the limit number?

Reply:
good point, given this is a library function, we need not log it here. 

But I will log it in ReceiveVideoStreamIndexTask.ts before we actually truncate it.